lib/spack/spack/mirror.py

```diff
@@ -60,17 +61,27 @@ def mirror_archive_filename(spec, fetcher, resourceId=None):
                 raise MirrorError(
                     "%s version does not specify an extension" % spec.name +
                     " and could not parse extension from %s" % fetcher.url)
+            # If the archive name is different to the package name
+            if spec.package.name != url.parse_name(fetcher.url):
+                name = url.parse_name(fetcher.url)
```

Comment:
It was my hack to get around issue #4342. Where it was saving the gem with the `.None` suffix.